src/components/Signup.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import Link from '@mui/material/Link';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import ToggleButton from '@mui/material/ToggleButton';
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+import { createUserWithEmailAndPassword } from '../api/auth';
+
+import { createFirebaseDao } from '../api/dao';
+
+function Signup() {
+    const navigate = useNavigate();
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [password, setPassword] = useState('');
+    const [email, setEmail] = useState('');
+    const [address, setAddress] = useState('');
+    const [userType, setUserType] = useState('senior');
+
+    const validateInput = () => {
```

Comment:
Minor comment. This function can be simplified if you do:

```
const validateInput = () => {
   return !(password.trim() === '' || password.length < 8)
}
```

Reply:
I just realized. Would it be better if we extract this `function validatePassword(password: string)` into a separate `utils` file and then add unit tests?

Personally, it would be better so at least we get to show off more code being unit tested for our project